### DISCUSSION TITLE
Does Native Control (no DCC) Still Work in macOS 15.0 Sequoia?

### DISCUSSION THREAD

**Discussion by l33** [MonitorControl/MonitorControl#1710] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Question

So far I have been using MonitorControl 4.2.0 with macOS 14 Sonoma and was quite happy using the native control to control the brightness of multiple LG Ultrafine displays at once. With the update to macOS 15.0 Sequoia I also had to update MonitorControl because it kept crashing. With its latest 4.3.3 release native brightness control no longer seems to work. I have to use DCC and I am not too happy about it:

1. DCC seems to use a range of brightness, for example it allows me to completely turn off the display which I do not want.
2. Every time my Mac wakes from sleep the brightness is set to whatever the native control was set to before changing the brightness using DCC. I effectively have to change the brightness again after every wake.

Is the brightness control using the native interface broken in macOS 15.0 Sequoia or am I overlooking something? Was the fix for 4.2.0 effectively to disable native brightnes control?

**Comment by waydabber** [comment, marked as answer]
No, everything works fine with Sequoia, except if your display is in HDR mode. When it is, the app will treat Sequoia's SDR peak brightness remap control as native brightness control and use that instead of DDC. If you want to continue using HDR mode and DDC at the same time, you need to switch to BetterDisplay. Make sure you enable DDC in HDR mode under the display's settings.

**Reply by l33** [reply]
OK, that does not seem to be what I am observing. The LG UltraFine does not have HDR and I am on a Mac Studio. There is not even an HDR setting in Display settings. The following screen shot shows the difference:
![image](https://github.com/user-attachments/assets/02401b42-902b-4e52-bf7c-a71f986b151d)

I will be away the next two weeks but I can have a deeper look into what's going on here after that.

**Reply by waydabber** [reply]
Ok, but DDC control is what you want, right? So this seems to be fine. If not, then simply turn off "Use hardware DDC control" and the app will not use DDC.

**Reply by l33** [reply]
> Ok, but DDC control is what you want, right? So this seems to be fine. If not, then simply turn off "Use hardware DDC control" and the app will not use DDC.

I want to use "Hardware (Apple)", not DCC. If I turn off "Use hardware DCC control" I get software control. This is, however, only in macOS 15. I tested MonitorControl 4.3.3 on macOS 14 and it does work as expected providing "Hardware (Apple)" control. At first sight it seems native control is no longer possible in macOS 15 but I guess I will have to  dig deeper.

**Reply by trigger2k20** [reply]
I can confirm in Sequoia 15.2 (24C101) the brightness works but not the sound (what I need desperately). It shows the change of volume/muting in the app and OSD when pressing F10-12 but It does not mute or change the volume. I use a m1 Mac mini with Lenovo LEN T27h-20.

**Reply by l33** [reply]
> I can confirm in Sequoia 15.2 (24C101) the brightness works [...]

@trigger2k20 can you confirm that you are using "Hardware (Apple)" control and not "Hardware (DCC)" as shown in the screenshots above?

**Reply by waydabber** [reply]
If you see "Hardware (Apple)" then your display might be in HDR mode. With Sequoia Apple provides a kind of brightness control (SDR peak brightness control) which MonitorControl interprets as native hardware control and switches to that instead of DDC. The side-effect of this is the lack of volume control. You can use BetterDisplay where this is managed better (the two control methods work together, keeping DDC for non-brightness + you have to option in Settings to switch to DDC instead of the native SDR peak brightness control).

**Reply by trigger2k20** [reply]
> > I can confirm in Sequoia 15.2 (24C101) the brightness works [...]
> 
> @trigger2k20 can you confirm that you are using "Hardware (Apple)" control and not "Hardware (DCC)" as shown in the screenshots above?

I can confirm that I see Hardware (DCC) I never saw Hardware (Apple).

![Bildschirmfoto 2025-01-02 um 20 11 36](https://github.com/user-attachments/assets/43f375a4-b449-443a-bd94-79f091fe2a08)

**Reply by trigger2k20** [reply]
> If you see "Hardware (Apple)" then your display might be in HDR mode. With Sequoia Apple provides a kind of brightness control (SDR peak brightness control) which MonitorControl interprets as native hardware control and switches to that instead of DDC. The side-effect of this is the lack of volume control. You can use BetterDisplay where this is managed better (the two control methods work together, keeping DDC for non-brightness + you have to option in Settings to switch to DDC instead of the native SDR peak brightness control).
<img width="597" alt="Bildschirmfoto 2025-01-02 um 21 18 53" src="https://github.com/user-attachments/assets/b1320456-b1ad-4180-81d8-d493f5eb2898" />


Well I tried to do but BetterDisplay brought the same results. What BD also said was my monitor does not support volume control via DCC :-( 

Guess i have to stick with Sound Control https://staticz.com/soundcontrol/

**Comment by l33** [comment]
@waydabber I found the issue for me. The detection of native control was broken by https://github.com/MonitorControl/MonitorControl/commit/e4af9a55a31a9cc8b22f2e176fecdf17e0c107bd. See my comment to the particular if block https://github.com/MonitorControl/MonitorControl/commit/e4af9a55a31a9cc8b22f2e176fecdf17e0c107bd#r151202126. Removing the if block fixes hardware control detection and control works again as expected.

**Reply by waydabber** [reply]
For HDR the app uses Sequoia's peak SDR brightness remap control (in Control Center) to configure brightness (however as a side effect, since the app is made in a way that it either uses Apple or DDC, all other DDC controls become unavailable in HDR mode). If you'd like a better support for HDR displays in Sequoia, I recommend switching to BetterDisplay (here DDC and Apple Control can work intermixed + you can force DDC control for brightness in HDR mode instead of the peak SDR remap control).
